controllers/ControllerComment.php

```diff
@@ -3,17 +3,72 @@
 namespace OC_Blog\Controllers;
 
 use OC_Blog\Models\CommentsManager;
+use OC_Blog\Tools\Session;
+
 
 class ControllerComment {
 
-	private $manager;
+	private $params;
+	private $method;
+	private $twig;
+	private $commentManager;
+
+
+	public function __construct($method, $twig, $params){
+		$this->method = $method;
+		$this->twig = $twig;
+		$this->params = $params;
+		$this->commentManager = new CommentsManager();
+		$target = $method[2];
+		if (method_exists(ControllerComment::class, $target) ) {
+			$this->$target();
+		}else{
+			echo $this->twig->render('404.twig');
+		}
+	}
+
+	public function addCommentPost(){
+		$key = (new Session)->getKey('user');
 
-	public function __construct() {
 
-		$this->manager = new CommentsManager();
+		if (!isset($key)){
+			header("Location: http://".$_SERVER['SERVER_NAME']."/Auth/login");
+			exit();
```

Comment:
Codacy found an issue: [The method addCommentPost() contains an exit expression.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=5982252)